service/turn.rest/src/main/java/org/kaazing/gateway/service/turn/rest/DefaultCredentialsGeneratorImpl.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright 2007-2016, Kaazing Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kaazing.gateway.service.turn.rest;
+
+import java.security.cert.Certificate;
+import java.time.Instant;
+import java.util.Base64;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import javax.security.auth.Subject;
+
+public class DefaultCredentialsGeneratorImpl implements TurnRestCredentialsGenerator {
```

Comment:
DefaultCredentialsGenerator is a better name, and this will be public API so it needs JavaDoc on what it does.  

We will then need to build and pull just this JavaDoc for this class into the API.  I'm not sure of the best way to do this, but some options are

1) Move class into its own module where we can build the java doc for it (probably over kill)
2) Configure javadoc plugin to just build java doc for this class (this seems ideal to me, though I'm not sure if that can be configurable)
3) Other options I haven't thought of